state/protocol/badger/state.go

```diff
@@ -433,131 +437,95 @@ func (state *State) bootstrapStatePointers(root protocol.Snapshot) func(*badger.
 			return fmt.Errorf("could not index sealed block: %w", err)
 		}
 
+		// insert first-height indices for epochs which have started
+		hasPrevious, err := protocol.PreviousEpochExists(root)
+		if err != nil {
+			return fmt.Errorf("could not check existence of previous epoch: %w", err)
+		}
+		if hasPrevious {
+			err = indexFirstHeight(root.Epochs().Previous())(tx)
+			if err != nil {
+				return fmt.Errorf("could not index previous epoch first height: %w", err)
+			}
+		}
+		err = indexFirstHeight(root.Epochs().Current())(tx)
+		if err != nil {
+			return fmt.Errorf("could not index current epoch first height: %w", err)
+		}
+
 		return nil
 	}
 }
 
 // bootstrapEpoch bootstraps the protocol state database with information about
 // the previous, current, and next epochs as of the root snapshot.
 // This has to be bootstrapped before dynamic protocol state.
```

Comment:
I find this comment confusing:
* we are hading an instance of `DynamicProtocolState` as input to the method. 
* But the method's documentation states that we have to bootstrap the Epoch before the `DynamicProtocolState`

🤯 

```suggestion
```

Reply:
My interpretation of the comment is that we need to run `bootstrapEpoch` before running `bootstrapProtocolState`, while initializing the _database_. The input instance of `DynamicProtocolState` is taken from the root state snapshot, which we have before bootstrapping any aspect of the database.

However, I'm not sure _why_ we need to run `bootstrapEpoch` before running `bootstrapProtocolState`. `bootstrapProtocolState` is just inserting indexes - I tried reversing the order and all tests passed.

I'll keep the order as it is currently, but remove that comment anyway as it appears to be outdated.

Reply:
If I remember correctly I wrote this because we need to have epoch events stored in order to query them during `RichProtocolStateEntry` construction, but code has evolved and we no longer do this, but the primary reason was to ensure that referenced epoch events could be queried from storage when we are bootstrapping dynamic protocol state.